### PR TITLE
Setup/fixes

### PR DESCRIPTION
**Pull request by edwardGr**
In this branch I have slightly fixed project structure by:

- Moving entities to more meaningful packages, modifying their names by adding 'Entity' word at the end 
- Changing project's name, groupId and artifactId. 
- Performing code formatting at Gender enum according to conventions.

### CODE REVIEW

File: src/main/java/com/george/messenger/domain/dao/entity/ConversationEntity.java [hunk 1]
```diff
@@ -24,25 +24,25 @@ public void setId(Long id) {
...
 
-    public void setParticipants(Set<User> participants) {
+    public void setParticipants(Set<UserEntity> participants) {
         this.participants = participants;
     }
 
     @Override
     public String toString() {
-        return "Conversation{" +
+        return "ConversationEntity{" +
```
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/ConversationEntity.java]: where's your overriding of equals & hashCode???

File: src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ public void setDateSent(Date dateSent) {
 
     @Override
     public String toString() {
-        return "Message{" +
+        return "MessageEntity{" +
```
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java]: same

File: src/main/java/com/george/messenger/domain/dao/entity/UserEntity.java [hunk 3]
```diff
@@ -95,17 +95,17 @@ public void setAge(Integer age) {
...
 
-    public void setConversationSet(Set<Conversation> conversationSet) {
+    public void setConversationSet(Set<ConversationEntity> conversationSet) {
         this.conversationSet = conversationSet;
     }
 
     @Override
     public String toString() {
-        return "User{" +
+        return "UserEntity{" +
```
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/UserEntity.java]: same

File: src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java [hunk 4]
```diff
@@ -24,11 +24,11 @@ public void setId(Long id) {
         this.id = id;
     }
 
-    public User getAuthor() {
+    public UserEntity getAuthor() {
         return author;
     }
 
-    public void setAuthor(User author) {
+    public void setAuthor(UserEntity author) {
```
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java]: use Builder pattern in your setters

File: src/main/java/com/george/messenger/domain/dao/entity/UserEntity.java [hunk 5]
```diff
@@ -34,9 +34,9 @@
...
 
     @ManyToMany(mappedBy = "participants")
-    private Set<Conversation> conversationSet;
+    private Set<ConversationEntity> conversationSet;
 
-    public User() {
+    public UserEntity() {
     }
 
     public String getUuid() {
```
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/UserEntity.java]: not uuid
uuid - just strategy, but field is 'id'

File: src/main/resources/hibernate.cfg.xml [hunk 6]
```diff
@@ -28,9 +28,9 @@
         <!-- Drop and re-create the database schema on startup -->
         <property name="hbm2ddl.auto">create</property>
 
-        <mapping class="com.george.messengerapp.entity.User"/>
-        <mapping class="com.george.messengerapp.entity.Conversation"/>
-        <mapping class="com.george.messengerapp.entity.Message"/>
+        <mapping class="com.george.messenger.domain.dao.entity.UserEntity"/>
+        <mapping class="com.george.messenger.domain.dao.entity.ConversationEntity"/>
+        <mapping class="com.george.messenger.domain.dao.entity.MessageEntity"/>
```
Comment (VadymPolyanski) [inline src/main/resources/hibernate.cfg.xml]: xml isn't good way for config

File: src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+@Entity
+public class MessageEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @OneToOne
+    private UserEntity author;
```
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java]: OneToOne?

Comment (edwardGr) [inline src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java]: Fixing it in another branch. It should be ManyToOne, figured it out the moment you told me about this.

File: src/main/resources/hibernate.cfg.xml (hunk 6 above)
Comment (edwardGr) [inline src/main/resources/hibernate.cfg.xml]: Setting up spring environment, where I'll config Hibernate.

Comment (VadymPolyanski) [inline src/main/resources/hibernate.cfg.xml]: spring jpa configuration
use annotation 
@EnableTransactionManagement()
@EnableJpaRepositories()
and another

File: src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java (hunk 7 above)
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java]: It should be in this branch
I'll not approve this if you don't fix..

File: src/main/resources/hibernate.cfg.xml (hunk 6 above)
Comment (VadymPolyanski) [inline src/main/resources/hibernate.cfg.xml]: Ok, I've merged another PR. 
I'm waiting for your hibernate jpa configuration.

File: src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java (hunk 7 above)
Comment (VadymPolyanski) [inline src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java]: Ok, I've merged another PR. 
I'm waiting for your hibernate jpa configuration.

Comment (edwardGr) [inline src/main/java/com/george/messenger/domain/dao/entity/MessageEntity.java]: Ok. Done.

**Pull request merged**